packages/nodejs/src/diagnose.ts

```diff
@@ -112,30 +113,28 @@ export class DiagnoseTool {
   }
 
   private async validatePushApiKey() {
-    return new Promise((resolve, reject) => {
-      const config = this.#config.data
-      const params = new URLSearchParams({
-        api_key: config["pushApiKey"] || "",
-        name: config["name"] || "",
-        environment: config["environment"] || "",
-        hostname: config["hostname"] || ""
+    const config = this.#config.data
+    const url = new URL(`/1/auth`, config["endpoint"])
+    const transmitter = new Transmitter(url.toString())
+    let statusCode: number
+
+    await transmitter
+      .transmit()
+      .then(responseData => {
+        statusCode = responseData["status"]
       })
-      const url = new URL(`/1/auth?${params.toString()}`, config["endpoint"])
-      const options = { method: "POST" }
-
-      const requestModule = url.protocol == "http:" ? http : https
-      const request = requestModule.request(url, options, function (response) {
-        const status = response.statusCode
-        if (status === 200) {
-          resolve("valid")
-        } else if (status === 401) {
-          reject("invalid")
-        } else {
-          reject(`Failed to validate: status ${status}`)
-        }
+      .catch(responseData => {
+        statusCode = responseData["status"]
       })
-      request.write("") // Send empty body
-      request.end()
+
+    return new Promise((resolve, reject) => {
```

Comment:
It feels a bit weird to create and immediately return a promise here. You can also directly return `Promise.resolve("valid")` or `Promise.reject("invalid")` here without the overhead of the promise function.

You probably didn't add the resolve and reject to `transmitter.transmit()` that now `await`s, because both the `then` and `catch` fetch the statuscode?

Reply:
> It feels a bit weird to create and immediately return a promise here. You can also directly return Promise.resolve("valid") or Promise.reject("invalid") here without the overhead of the promise function.

I'll use those helpers as return values, thank you

> You probably didn't add the resolve and reject to transmitter.transmit() that now awaits, because both the then and catch fetch the statuscode?

Yep, that's correct